packages/commerce/src/machines/commerce/utils.ts

```diff
@@ -126,24 +101,28 @@ export const getPriceParams = (machineContext: CommerceMachineContext) => {
   }
   const {site, id: sellable_id, type} = sellable
 
+  const upgradeParams = {
+    upgrade_from_sellable_id: upgradeFromSellable?.slug,
+    upgrade_from_sellable: upgradeFromSellable?.type,
+  }
+
+  const sellableStuff: EggheadSellableParam = {
+    sellable_id,
+    sellable: type.toLowerCase(),
+    ...(!!upgradeParams && upgradeParams),
+    ...(!!quantity && {quantity}),
+  }
+
   return isEmpty(stripePriceId)
-    ? pickBy({
-        sellables: [
-          pickBy({
-            sellable_id,
-            upgrade_from_sellable_id: upgradeFromSellable?.slug,
-            upgrade_from_sellable: upgradeFromSellable?.type,
-            sellable: type.toLowerCase(),
-            quantity,
-          }),
-        ],
+    ? {
+        sellables: [sellableStuff],
         site: process.env.NEXT_PUBLIC_SITE_NAME || 'TEST_PRODUCT',
         code: appliedCoupon,
-      })
+      }
     : {id: stripePriceId}
 }
 
-export const eggheadPriceCheckUrl = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/sellable_purchases/prices`
+export const eggheadPriceCheckUrl = `/api/prices`
```

Comment:
local call for prices that is a proxy for this endpoint

generally, I want to migrate to this pattern for these projects where it makes sense.